andela/knights-bn-backend
# #170947672 comment on request

#### What does this PR do?
- Allow users to comment on their request.
#### Description of Task to be completed?
- A user may need to comment on his request to provide additional information 
- This will allow users to make comment on their risk
#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git checkout ft-comment-on-request-170947672 ` to switch to this branch.
- run `npm install` to install dependencies.

- create a .env file then add environment variable as shown in ` .env.example`,  check our slack channel `code-knights-12` for .env


- create two Postgres database for development and then test
- run `npm run createTables` to create tables for you

- run `npm run dev-start` to start the server.

- signup a regular user user by POST `localhost:4000/api/v1/auth/signup`
{
                 "firstName": "Mugenzi",
                  "lastName": "Jean",
                  "gender": "male",
                  "passportNumber": "ui8908dv",
                 "email": "david@andela.com",
                 "password": "Niyonkuru@1"
            }

- update your manager by  PATCH 'http://localhost:4000/api/v1/edit/user/profile'

- make a new request by POST `http://localhost:4000/api/v1/trips/returnTrip`

{
	"origin": "Burundo",
	"destination": "kgl",
	"departureDate": "9002-04-02",
	"returnDate": "9003-05-18",
	"accommodation": "XYZ campus",
	"reason": "Having a meeting",
	"passportNumber": "yuyut56"
}

-use the following to comment on newly created request `http://localhost:4000/api/v1/trips/comment?requestId=id`     replace id by the id of the request

- then to the see your comment go to `localhost:4000/api/v1/trips/myRequest` 


#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

- [#170947672](https://www.pivotaltracker.com/story/show/170947672)

#### Screenshots (if appropriate)
              

Create a request

<img width="935" alt="Screen Shot 2020-03-04 at 16 04 10" src="https://user-images.githubusercontent.com/48021034/75887066-e49b1a80-5e31-11ea-9fe6-a242d254be1a.png">



comment on the request

<img width="851" alt="Screen Shot 2020-03-04 at 16 04 18" src="https://user-images.githubusercontent.com/48021034/75887106-f2e93680-5e31-11ea-84d8-65f1256e1b0b.png">


view all my request with comment

<img width="577" alt="Screen Shot 2020-03-04 at 16 04 41" src="https://user-images.githubusercontent.com/48021034/75887143-00062580-5e32-11ea-8e36-c53224a2cecf.png">



#### Questions:
N/A
